lib/reply.js

```diff
@@ -292,5 +310,36 @@ function notFound (reply) {
 
 function noop () {}
 
+function getHeaderProper (reply, key) {
+  key = key.toLowerCase()
+  var res = reply.res
+  var value = reply._headers[key]
+  if (value === undefined && res.hasHeader(key)) {
```

Comment:
Using `res.hasHeader` doesn't seem like it would help much. It's almost the exact same code as `res.getHeader` so it would be simpler just to have one function that's the same as `getHeaderFallback`.

Reply:
I do not have investigated why, but `res.getHeader` slow us down, while `res.hasHeader` does not.

Reply:
Oh that is very interesting.